src/lib.rs

```diff
@@ -776,6 +778,8 @@ impl Format {
                 .unwrap(),
             Format::Dynamic(_name, _dynformat, f) => f.match_bounds(module),
             Format::Apply(_) => Bounds::at_least(1),
+            // FIXME - do we have any way of approximating this better?
+            Format::ForEach(_expr, _lbl, _f) => Bounds::any(),
```

Comment:
expr bounds * f bounds?

Reply:
Oh right yes, we would need bounds on sequences and we don't have that right now and won't for a while. The comment can probably say "can't do better without sequence length bounds" or similar.